### PR TITLE
FIx issue #464 corrupt log file

### PR DESCRIPTION
**Pull request by barry-scott**
Must only append the first line of the commit to the log
Otherwise the log file is not parsable by GitPython.

### CODE REVIEW

**Comment by Byron** [comment]
Do you think it's possible for you to write a test which goes red if the change is not there ? The [current test-suite](https://github.com/gitpython-developers/GitPython/blob/5c8ff218cb3ee5d3dd9119007ea8478626f6d2ce/git/test/test_reflog.py#L18) should be a good starting-point for a fixture-based test.

**Comment by barry-scott** [comment]
Please don't wait for me to write the test. I have a lot of work on my plate at moment.
Maybe merge this and leave the bug open awiting a test case?

How I detected the bug was:

1. Use GitPython to commit to a repo - causing the fault
2. Use GitPython to scan the log ref file - detecting the fault

**Comment by barry-scott** [comment]
I'm not sure how to write the tests you want.

Can you point to docs on what is required?

File: git/cmd.py [hunk 1]
```diff
@@ -629,7 +636,13 @@ def execute(self, command,
...
         def _kill_process(pid):
             """ Callback method to kill a process. """
-            p = Popen(['ps', '--ppid', str(pid)], stdout=PIPE)
+            if sys.platform == 'win32':
+                CREATE_NO_WINDOW = 0x08000000
+                creationflags = CREATE_NO_WINDOW
+            else:
+                creationflags = None
+
+            p = Popen(['ps', '--ppid', str(pid)], stdout=PIPE, creationflags)
```
Comment (Byron) [inline git/cmd.py]: I am afraid that using `creationflags` as positional argument doesn't actually end up in the correct spot. Also I thought that positional arguments after keyword args are not allowed.

File: git/cmd.py [hunk 2]
```diff
@@ -609,6 +609,12 @@ def execute(self, command,
         # end handle
 
         try:
+            if sys.platform == 'win32':
+                CREATE_NO_WINDOW = 0x08000000
```
Comment (Byron) [inline git/cmd.py]: `CREATE_NO_WINDOW` is used twice, is there a reason it is not a constant on module or class level ?

File: git/cmd.py [hunk 3]
```diff
@@ -609,6 +609,12 @@ def execute(self, command,
         # end handle
 
         try:
+            if sys.platform == 'win32':
```
Comment (Byron) [inline git/cmd.py]: Generally for these kinds of assignment, I believe it's easier to read by starting out like this:
```python
creationflags = None
if sys.platform == 'win32':
	creationflags = CREATE_NO_WINDOW
```

**Comment by Byron** [comment]
Now this PR contains two distinct and independent changes. The most recent one is hard to test, and I'd be fine without one.
As for the first set of changes, I believe I already referred to the respective test-suite. Generally, you want to write a test that fails until your changes are applied. This is the whole trick. The suite in question should already contain everything you need to setup your own.

File: git/cmd.py (hunk 1 above)
Comment (Byron) [inline git/cmd.py]: This concern is reflected in actual breakage as well: https://travis-ci.org/gitpython-developers/GitPython/jobs/148304657#L318

File: git/cmd.py (hunk 2 above)
Comment (barry-scott) [inline git/cmd.py]: I intended to provide a obvious patch. You usually change my process control patches so I left that for you to decide on where you wished to define CREATE_NO_WINDOW.

**Comment by barry-scott** [comment]
I fixed the keyword parameter passing.

**Comment by barry-scott** [comment]
I have added a basic test by changing one of the existing test_repo.py tests.

You can see the test fail by undoing the first line fix and running the test.

**Comment by Byron** [comment]
@barry-scott Thank you, looks very good now ! I am particularly excited about the test, and hope in future you can give *test-first* development a shot as well.

**Comment by barry-scott** [comment]
To get the test as one patch and the the test + fix as a second you will have to fix the dependency of the tests on master. Use of master means that you do not have a reproducible test. Its inputs depend on what the user did to create there master.

I suspect that if you had to create the test data that you use master for you might have found this bugs ages ago.

**Pull request merged**